### PR TITLE
Added `apply_fmt_fn(replace)` argument

### LINKED ISSUE #285: Update `apply_fmt_fn()` to not overwrite existing formatted statistics
In some cases, users want apply formatting functions across a statistic. in the example below, the rounding function ensures the rounded statistics sum to the same number as the raw stats.

Perhaps we could also introduce a `apply_fmt_fn(by)` argument, and the statistics would be applied to the group of statistics. My only reservation is that this would fail when there are a mix a formatting functions that get applied to vectors and one-in-one-out formatting function (which would apply to more complex cases). But in those cases, the users can easily program around it.

``` r
library(cards)

ex <-
  ard_categorical(mtcars, variables = cyl, statistic = ~"p") |> 
  dplyr::mutate(
    .by = "stat_name",
    stat_fmt = 
      if (stat_name[1] == "p") JLutils::round_preserve_sum(unlist(stat) * 100, digits = 2) |> as.list()
      else purrr::map(stat, ~NULL)
  )
#> Registered S3 method overwritten by 'GGally':
#>   method from   
#>   +.gg   ggplot2
ex
#> {cards} data frame: 3 x 10
#>   variable variable_level stat_name stat_label  stat stat_fmt
#> 1      cyl              4         p          % 0.344    34.37
#> 2      cyl              6         p          % 0.219    21.88
#> 3      cyl              8         p          % 0.438    43.75
#> ℹ 4 more variables: context, fmt_fn, warning, error

ex |> apply_fmt_fn()
#> {cards} data frame: 3 x 10
#>   variable variable_level stat_name stat_label  stat stat_fmt
#> 1      cyl              4         p          % 0.344     34.4
#> 2      cyl              6         p          % 0.219     21.9
#> 3      cyl              8         p          % 0.438     43.8
#> ℹ 4 more variables: context, fmt_fn, warning, error
```

<sup>Created on 2024-08-02 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

### PR DESCRIPTION
**Pull request by ddsjoberg**
**What changes are proposed in this pull request?**
* Added `apply_fmt_fn(replace)` argument. Use `replace=FALSE` to retain any previously formatted statistics in the `stat_fmt` column. (#285)

**Reference GitHub issue associated with pull request.** _e.g., 'closes #<issue number>'_
closes #285

--------------------------------------------------------------------------------

Pre-review Checklist (if item does not apply, mark is as complete)
- [x] **All** GitHub Action workflows pass with a :white_check_mark:
- [x] PR branch has pulled the most recent updates from master branch: `usethis::pr_merge_main()`
- [x] If a bug was fixed, a unit test was added.
- [x] Code coverage is suitable for any new functions/features (generally, 100% coverage for new code): `devtools::test_coverage()`
- [x] Request a reviewer

Reviewer Checklist (if item does not apply, mark is as complete)

- [x] If a bug was fixed, a unit test was added.
- [x] Run `pkgdown::build_site()`. Check the R console for errors, and review the rendered website.
- [x] Code coverage is suitable for any new functions/features: `devtools::test_coverage()`

When the branch is ready to be merged:
- [x] Update `NEWS.md` with the changes from this pull request under the heading "`# cards (development version)`". If there is an issue associated with the pull request, reference it in parentheses at the end update (see `NEWS.md` for examples).
- [x] **All** GitHub Action workflows pass with a :white_check_mark:
- [x] Approve Pull Request
- [x] Merge the PR. Please use "Squash and merge" or "Rebase and merge".

### CODE REVIEW

**Comment by ddsjoberg** [comment]
Hi @bzkrouse @Melkiades @ayogasekaram @edelarua ! I requested multiple reviewers, but we only need one. When you have the time to reivew, can you remove the other reviewers before you begin?

File: R/apply_fmt_fn.R [hunk 1]
```diff
@@ -5,31 +5,40 @@
 #'
 #' @param x (`data.frame`)\cr
 #'   an ARD data frame of class 'card'
+#' @param replace (scalar `logical`)\cr
+#'   logical indicating whether to replace values in the `'stat_fmt'` column (if present).
+#'   Default is `FALSE`
```
Comment (Melkiades) [inline R/apply_fmt_fn.R:10]:
```suggestion
#'   Default is `FALSE`.
```

**Review by Melkiades — APPROVED** [review]
Lgtm! Thanks Daniel ^^

**Comment by ddsjoberg** [comment]
Thank you so so much for the review @Melkiades ! 🕺🏼

**Pull request merged**